RecoTracker/MkFitCore/src/MkFinder.h

```diff
@@ -0,0 +1,351 @@
+#ifndef RecoTracker_MkFitCore_src_MkFinder_h
+#define RecoTracker_MkFitCore_src_MkFinder_h
+
+#include "MkBase.h"
+#include "RecoTracker/MkFitCore/interface/TrackerInfo.h"
+#include "RecoTracker/MkFitCore/interface/Track.h"
+
+#include "RecoTracker/MkFitCore/interface/HitStructures.h"
+
+// Define to get printouts about track and hit chi2.
+// See also MkBuilder::backwardFit().
+
+//#define DEBUG_BACKWARD_FIT_BH
+//#define DEBUG_BACKWARD_FIT
+
+namespace mkfit {
+
+  class CandCloner;
+  class CombCandidate;
+  class LayerOfHits;
+  class FindingFoos;
+  class IterationParams;
+  class IterationLayerConfig;
+  class SteeringParams;
+
+  //#ifdef DEBUG_BACKWARD_FIT
+  //class Event;
+  //#endif
+
+#ifdef DUMPHITWINDOW
+  class Event;
+#endif
+
+  // NOTES from MkFitter ... where things were getting super messy.
+  //
+  // Would like to get rid of msErr, msPar arrays ... single one should do.
+  // Would also like to get rid of Nhits ... but this is a tougher one.
+  // Fitting is somewhat depending on this (not the new inter-slurp version).
+  // No need to have hit array as matriplex. Ideally, don't even want the full
+  // array if we could copy / clone it from the original candidate.
+  // Copy as needed or copy full size (compile time constant)? memcpy, std::copy
+  // Do need per track size.
+  //
+  // Can also have nFoundHits ... then do not need countValid/InvalidHits().
+  //
+  // For now will prefix the new variables with 'tf' for track finding.
+  // Changes will go into all finding functions + import-with-hit-idcs.
+  //
+  // Actually ... am tempted to make MkFinder :)
+
+  class MkFinder : public MkBase {
+  public:
+    static constexpr int MPlexHitIdxMax = 16;
+
+    using MPlexHitIdx = Matriplex::Matriplex<int, MPlexHitIdxMax, 1, NN>;
+    using MPlexQHoT = Matriplex::Matriplex<HitOnTrack, 1, 1, NN>;
+
+    //----------------------------------------------------------------------------
+
+    MPlexQF Chi2;
```

Comment:
use CMS naming convention for data members

Reply:
Addressed in Round 4 follow-up PR trackreco/cmssw#79 commit: https://github.com/trackreco/cmssw/pull/79/commits/85c149fb2b8af53d752e1a334f4d90feed9ba2ef